tests/test_matrix_utils_relay.py

```
@@ -1334,9 +1333,99 @@ async def run_in_executor(self, _executor, func, *args):
+        assert "reacted" in queued_kwargs["text"]
+        assert queued_args[0].__name__ == "send_text_reply"
+
+
+async def test_on_room_message_reaction_non_numeric_meshtastic_id(
+    mock_room, test_config
+):
+    """Non-numeric mapped meshtastic_id should send as normal message, not reply."""
+    from nio import ReactionEvent
+
+    class MockReactionEvent(ReactionEvent):
+        def __init__(self, source, sender, server_timestamp):
```

Ruff (0.15.12)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)